.github/workflows/build-voyager.yml

```diff
@@ -0,0 +1,69 @@
+name: Create yb-voyager binary
+
+on:
+  create:
+    tags:
+      - 'yb-voyager/v*' # Trigger on any tag starting with 'yb-voyager/v'
+
+jobs:
+  build:
+    name: Build on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    permissions:
+      contents: write
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: "1.20"
+
+      - name: Build Binary
+        run: |
+          cd yb-voyager
+          if [ ${{ matrix.os }} == "ubuntu-latest" ]; then
+            GOOS=linux GOARCH=amd64 go build -o voyager-linux-amd64
+          elif [ ${{ matrix.os }} == "macos-latest" ]; then
+            CGO_ENABLED=1 GOOS=darwin GOARCH=amd64 go build -o voyager-darwin-amd64
+            CGO_ENABLED=1 GOOS=darwin GOARCH=arm64 go build -o voyager-darwin-arm64
+          fi
+
+      - name: Archive the files
+        run: |
+          cd yb-voyager
+          if [ ${{ matrix.os }} == "ubuntu-latest" ]; then
+            mkdir voyager-linux-amd64-package
```

Comment:
Drop the `-package` suffix from the name. Prefix the binary files with `yb-` to avoid conflict with the directory.